tests/bluetooth/bsim/host/adv/encrypted/css_sample_data/src/central.c

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "common.h"
+
+extern const uint8_t *session_key;
+extern const uint8_t *iv;
+extern uint8_t size_sample_ad_data;
+extern const uint8_t *sample_ad_data;
+extern uint8_t size_sample_ead;
+extern const uint8_t *sample_ead;
+
+extern int data_set;
+
+static bool data_parse_cb(struct bt_data *data, void *user_data)
+{
+	if (data->type == BT_DATA_ENCRYPTED_AD_DATA) {
+		int err;
+		size_t decrypted_data_size = BT_EAD_DECRYPTED_PAYLOAD_SIZE(data->data_len);
+
+		if (decrypted_data_size != size_sample_ad_data) {
+			LOG_ERR("Size of decrypted data: %d", decrypted_data_size);
+			LOG_ERR("Size of sample data: %d", size_sample_ad_data);
+			FAIL("Computed size of data does not match the size of the data from the "
+			     "sample. (data set %d)\n",
+			     data_set);
+		}
+
+		struct net_buf_simple *decrypted_buf = k_malloc(sizeof(struct net_buf_simple));
+
+		decrypted_buf->size = decrypted_data_size;
```

Comment:
don't think you have to use a runtime alloc. Since you already know the size (CSS data) then you can create a static one with `NET_BUF_SIMPLE_DEFINE_STATIC(_name, _size)`. If you want to keep the runtime alloc, I would advise to at least move it to its own fn like so:
```c
static struct net_buf_simple* alloc_buf(size_t size)
{
	struct net_buf_simple *buf = k_malloc(sizeof(struct net_buf_simple));

	buf->size = size;
	buf->__buf = k_malloc(sizeof(buf->data[0]) * size);
	net_buf_simple_reset(buf);

	return buf;
}

static void free_buf(struct net_buf_simple *buf)
{
	k_free(buf->__buf);
	k_free(buf);
}
```

Reply:
Agreed. That way you can also remove the heap if you want.